capstone-ui/src/Components/ResultsBox/ResultsBox.jsx

```diff
@@ -97,6 +99,11 @@ export default function ResultsBox( { locationLevel, countyFIPS, selectedCandida
               ))}
             </tbody>
           </table>
+          {locationLevel != 3 && candidatesShowing < candidates.length && (
+            <button onClick={() => setCandidatesShowing((prev) => prev + initialCandidates)}>
```

Comment:
I'm not sure if `.slice` on line 90 throws out of bounds exception.

My concern in this line is you are incrementing by the page size of 5. Let's take an example of 12 candidates in total. Clicking the show more once will increment `candidatesShowing = 10`. `10 < 12`, so `Show More` will still display. Clicking on this will increment yet again `candidatesShowing = 15`. Now, `15 < 12` and we don't display `Show More` which is expected and works. However, on line 90, we will do `candidates.slice(0, 15)` given the state update. This has potential to cause exceptions if the api method `slice` does not gracefully handle this and uses the array size instead of the higher value. If it does, I think it would still be best practice to ensure we don't run into this issue.

Reply:
I've checked every state and it doesn't go out of bounds on any of them, but I will take a closer at what exactly it's doing.